core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala

```diff
@@ -49,42 +44,12 @@ private[external] trait ExternalBackendUtils extends SharedBackendUtils {
     */
   override def getH2OClientArgs(conf: H2OConf): Seq[String] = {
     new ArgumentBuilder()
-      .add("-flatfile", conf.h2oCluster.map(clusterStr => SharedBackendUtils.saveFlatFileAsFile(clusterStr).getAbsolutePath))
+      .add("-flatfile", conf.h2oCluster.map(clusterStr => saveFlatFileAsFile(clusterStr).getAbsolutePath))
       .add(super.getH2OClientArgs(conf))
       .addIf("-watchdog_client", conf.isAutoClusterStartUsed)
       .buildArgs()
   }
 
-  /** Check Spark and H2O environment, update it if necessary and and warn about possible problems.
-    *
-    * This method checks the environments for generic configuration which does not depend on particular backend used
-    * In order to check the configuration for specific backend, method checkAndUpdateConf on particular backend has to be
-    * called.
-    *
-    * This method has to be called at the start of each method which override this one
-    *
-    * @param conf H2O Configuration to check
-    * @return checked and updated configuration
-    * */
-  override def checkAndUpdateConf(conf: H2OConf): H2OConf = {
-    super.checkAndUpdateConf(conf)
```

Comment:
Move to checkAndUpdateConf on ExternalH2OBackend, this was unnecessary override. We just need to override it on ExternalH2OBackend and call super to ensure we call the method in SharedBackendUtils.

Same holds for InternalH2OBackend